File: src/main/java/org/taktik/icure/services/external/rest/v1/dto/embed/CareTeamMembershipDto.java
package org.taktik.icure.services.external.rest.v1.dto.embed;

import org.taktik.icure.entities.embed.MembershipType;

import java.io.Serializable;

public class CareTeamMembershipDto implements Serializable {
    private Long startDate;
    private Long endDate;
    private String careTeamMemberId;
    private MembershipType membershipType;

    public Long getStartDate() { return startDate; }

    public void setStartDate(Long startDate) { this.startDate = startDate; }

    public Long getEndDate() { return endDate; }

    public void setEndDate(Long endDate) { this.endDate = endDate; }

    public String getCareTeamMemberId() { return careTeamMemberId;}

    public void setCareTeamMemberId(String careTeamMemberId) { this.careTeamMemberId = careTeamMemberId; }

    public MembershipType getMembershipType() { return membershipType;}

    public void setMembershipType(MembershipType membershipType) { this.membershipType = membershipType; }
}

File: src/main/kotlin/org/taktik/icure/services/external/rest/v1/dto/embed/CareTeamMembershipDto.kt
package org.taktik.icure.services.external.rest.v1.dto.embed

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.github.pozo.KotlinBuilder
import java.io.Serializable

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@KotlinBuilder
data class CareTeamMembershipDto(
        val startDate: Long? = null,
        val endDate: Long? = null,
        val careTeamMemberId: String? = null,
        val membershipType: MembershipTypeDto? = null,
        override val encryptedSelf: String? = null
) : EncryptedDto, Serializable
